core/mempool/src/lib.rs

```diff
@@ -99,23 +201,40 @@ impl Pool {
             self.receipts.write().expect(POISONED_LOCK_ERR).remove(receipt);
         }
     }
+
+    pub fn add_pending(&self, authority_id: AuthorityId, hash: CryptoHash) {
+        self.pending_snapshots.write().expect(POISONED_LOCK_ERR).push((authority_id, hash))
+    }
+
+    pub fn add_payload_snapshot(
+        &self,
+        authority_id: AuthorityId,
+        payload: ChainPayload,
+    ) -> Result<(), String> {
+        // TODO: payload should be diff and then calculate the actual payload, but for now it is full.
+        let h = hash_struct(&payload);
```

Comment:
I think we use `CryptoHash::default()` as hash of empty payload, everywhere in this code. This line seems to be inconsistent with that as it actually computes some hash on `payload`, if `payload` is empty this hash is going to be different from `CryptoHash::default()`.